### PR TITLE
test(command-dev): limit concurrency

### PR DESCRIPTION
**Pull request by erezrokah**
This is an attempt to make our CI more stable by limiting the concurrency of the servers we spawn during tests.

### CODE REVIEW

**Comment by erezrokah** [comment]
@ehmicky this doesn't solve all our CI issues, but looks like it doesn't impact performance:
https://github.com/netlify/cli/actions/runs/477184868
https://github.com/netlify/cli/actions/runs/477774268

File: tests/command.dev.test.js [hunk 1]
```diff
@@ -14,6 +15,9 @@ const { withDevServer } = require('./utils/dev-server')
 const { startExternalServer } = require('./utils/external-server')
 const { withSiteBuilder } = require('./utils/site-builder')
 
+const CONCURRENCY = 1
+const semaphore = process.env.CI === 'true' ? new Semaphore(CONCURRENCY) : null
```
Comment (ehmicky) [inline tests/command.dev.test.js:19]: Would the following work too?

```js
const testFunc = process.env.CI === 'true' ? test.serial.bind(test) : test

testFunct('test title', (t) => { ... })
```

Comment (erezrokah) [reply tests/command.dev.test.js:19]: You're right, I was playing with different values or `CONCURRENCY` (`4->2->1`).
We can remove the dependency and enforce `serial`.

Comment (erezrokah) [reply tests/command.dev.test.js]: I chose to do it a bit different. It required disabling a lint rule, but makes the spec consistent with other ones.

**Review by ehmicky — APPROVED** [review]

**Pull request merged**